### PR TITLE
Adding ConnectorHandler for SubstraitVeloxPlanConverter

### PR DESCRIPTION
**Pull request by vibhatha**
At present, the `SubstraitVeloxPlanConverter` utilizes a [hard-coded](https://github.com/facebookincubator/velox/blob/main/velox/substrait/SubstraitToVeloxPlan.cpp#L412) implementation of `HiveTableHandle` and `HiveColumnHandle`. To increase modularity and support diverse use cases, the introduction of a `ConnectorHandler` is proposed, which enables the generation of customized `ConnectorTableHandle` and `ColumnHandle` instances. This would "allow for connectors other than Hive".

### CODE REVIEW

**Comment by vibhatha** [comment]
cc @mbasmanova

File: velox/substrait/SubstraitToVeloxPlan.cpp [hunk 1]
```diff
@@ -409,28 +439,15 @@ core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
     }
   }
 
-  // Do not hard-code connector ID and allow for connectors other than Hive.
-  static const std::string kHiveConnectorId = "test-hive";
-
   // Velox requires Filter Pushdown must being enabled.
   bool filterPushdownEnabled = true;
-  std::shared_ptr<connector::hive::HiveTableHandle> tableHandle;
+  std::shared_ptr<connector::ConnectorTableHandle> tableHandle;
```
Comment (mbasmanova) [inline velox/substrait/SubstraitToVeloxPlan.cpp:444]: The logic above is Hive-connector specific.

> case SubstraitFileFormatCase::kParquet:

File: velox/substrait/SubstraitToVeloxPlan.h [hunk 2]
```diff
@@ -22,11 +22,46 @@
 
 namespace facebook::velox::substrait {
 
+class ConnectorHandler {
+ public:
+  virtual ~ConnectorHandler() = default;
+
+  virtual std::shared_ptr<connector::ConnectorTableHandle> createTableHandle(
+      connector::hive::SubfieldFilters filters = {}) = 0;
```
Comment (mbasmanova) [inline velox/substrait/SubstraitToVeloxPlan.h:30]: SubfieldFilters are HiveConnector-specific.

File: velox/substrait/SubstraitToVeloxPlan.h [hunk 3]
```diff
@@ -22,11 +22,46 @@
...
+      const TypePtr& columnHandleDataType) = 0;
+};
+
+class HiveConnectorHandler : public ConnectorHandler {
+ public:
+  HiveConnectorHandler(
+      const std::string& connectorId,
+      const std::string& tableName,
+      const bool filterPushDownEnabled,
+      const connector::hive::HiveColumnHandle::ColumnType& columnType);
```
Comment (mbasmanova) [inline velox/substrait/SubstraitToVeloxPlan.h:43]: Column types are per column, not per connector, no?

File: velox/substrait/SubstraitToVeloxPlan.cpp [hunk 4]
```diff
@@ -20,6 +20,36 @@
...
+  return std::make_shared<connector::hive::HiveTableHandle>(
+      connectorId_,
+      tableName_,
+      filterPushDownEnabled_,
+      std::move(filters),
+      nullptr);
+}
+
+std::shared_ptr<connector::ColumnHandle>
+HiveConnectorHandler::createColumnHandle(
```
Comment (mbasmanova) [inline velox/substrait/SubstraitToVeloxPlan.cpp:46]: How would we handle queries that access both regular columns and partition keys?

File: velox/substrait/tests/FunctionTest.cpp [hunk 5]
```diff
@@ -40,8 +40,18 @@ class FunctionTest : public ::testing::Test {
   std::shared_ptr<vestrait::SubstraitParser> substraitParser_ =
       std::make_shared<vestrait::SubstraitParser>();
 
+  std::shared_ptr<ConnectorHandler> getConnector() {
+    return std::make_shared<HiveConnectorHandler>(
+        "function-test-hive",
+        "function-hive-table",
```
Comment (mbasmanova) [inline velox/substrait/tests/FunctionTest.cpp:46]: It is a bit strange to hard-code table name here. Shouldn't it come from the Substrait plan? How would we process queries that read from multiple tables (union, join)?

**Review by mbasmanova — COMMENTED** [review]
@vibhatha This change is a bit difficult for me to understand. Would you create a GitHub Issue and describe overall design in there? Specifically, I'd like to understand how TableScan node is represented in Substrait and how it can be mapped to TableScan node in Velox.

File: velox/substrait/SubstraitToVeloxPlan.h (hunk 2 above)
Comment (vibhatha) [reply velox/substrait/SubstraitToVeloxPlan.h:30]: You're correct, I need to generalize this.

**Comment by vibhatha** [comment]
cc @mbasmanova I added a proposal for this task [here](https://github.com/facebookincubator/velox/issues/4561).

File: velox/substrait/SubstraitToVeloxPlan.h [hunk 6]
```diff
@@ -22,11 +22,73 @@
 
 namespace facebook::velox::substrait {
 
+class SubstraitToVeloxFilter {
+ public:
+  virtual void addFilter(
+      const std::string& filterName,
+      std::unique_ptr<common::DoubleRange>&& range) = 0;
```
Comment (mbasmanova) [inline velox/substrait/SubstraitToVeloxPlan.h:29]: Why are we hard-coding DoubleRange here? There are other types of filters that need to be supported as well. Should we use common::Filter instead?

Comment (vibhatha) [reply velox/substrait/SubstraitToVeloxPlan.h:29]: Good point. I will look into this.

**Pull request closed without merging**